### PR TITLE
Site-to-site vpn api docs

### PR DESCRIPTION
**Pull request by franzpgarcia**

### CODE REVIEW

File: source/includes/cloudstack/_s2s_vpns.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+### Site-to-site VPN
+
```
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: Should add a description

File: source/includes/cloudstack/_s2s_vpns.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+Retrieve a list of all site-to-site VPNs in an [environment](#environments)
+
+Attributes | &nbsp;
+---------- | -----
+`id`<br/>*UUID* | The id of the site-to-site VPN
+`name`<br/>*string* | The name of the site-to-site VPN
+`state`<br/>*string* | The state of the site-to-site VPN. Can be Connected, Pending or Disconnected. If disconnected, you can try to use the [reset](#reset-a-site-to-site-vpn) operation
+`vpcId`<br/>*UUID* | The VPC for which the site-to-site VPN was created.
+`gateway`<br/>*string*  | The gateway of the network you want to connect to. NOTE: you cannot use a gateway that has already been used by a site-to-site VPN in your environment
+`cidr`<br/>*string*  | Cidr of the network you want to connect to.
```
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: CIDR

File: source/includes/cloudstack/_s2s_vpns.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+`ipSecPsk`<br/>*string*  | IPSec pre-shared key. Mmust contain at least 10 alphanumeric characters.
```
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: Typo

File: source/includes/cloudstack/_s2s_vpns.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+`ikeEncryptionAlgorithm`<br/>*string*  | The Internet Key Exchange (IKE) policy for phase-1. The supported encryption algorithms are AES128, AES192, AES256, and 3DES.
+`ikeHashAlgorithm`<br/>*string*  | The IKE hash for phase-1. The supported hash algorithms are SHA1 and MD5.
+`ikeDhGroup`<br/>*string*  | A public-key cryptography protocol which allows two parties to establish a shared secret over an insecure communications channel. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
+`ikeLifetime`<br/>*integer*  | The phase-1 lifetime of the security association in seconds.
+`espEncryptionAlgorithm`<br/>*string*  | Encapsulating Security Payload (ESP) algorithm within phase-2. The supported encryption algorithms are AES128, AES192, AES256, and 3DES.
+`espHashAlgorithm`<br/>*string*  | Encapsulating Security Payload (ESP) hash for phase-2. Supported hash algorithms are SHA1 and MD5.
+`espPerfectForwardSecrecy`<br/>*string*  |  Forward Secrecy (or PFS) is the property that ensures that a session key derived from a set of long-term public and private keys will not be compromised. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
```
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: just Forward Secrecy?

File: source/includes/cloudstack/_s2s_vpns.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+```
+
+<code>GET /services/<a href="#service-connections">:service_code</a>/<a href="#environments">:environment_name</a>/sitetositevpns</code>
+
+Retrieve a list of all site-to-site VPNs in an [environment](#environments)
+
+Attributes | &nbsp;
+---------- | -----
+`id`<br/>*UUID* | The id of the site-to-site VPN
+`name`<br/>*string* | The name of the site-to-site VPN
```
Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: We should add the same precision about unicity within the environment here too

File: source/includes/cloudstack/_s2s_vpns.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+`ikeDhGroup`<br/>*string*  | A public-key cryptography protocol which allows two parties to establish a shared secret over an insecure communications channel. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
+`ikeLifetime`<br/>*integer*  | The phase-1 lifetime of the security association in seconds.
+`espEncryptionAlgorithm`<br/>*string*  | Encapsulating Security Payload (ESP) algorithm within phase-2. The supported encryption algorithms are AES128, AES192, AES256, and 3DES.
+`espHashAlgorithm`<br/>*string*  | Encapsulating Security Payload (ESP) hash for phase-2. Supported hash algorithms are SHA1 and MD5.
+`espPerfectForwardSecrecy`<br/>*string*  |  Forward Secrecy (or PFS) is the property that ensures that a session key derived from a set of long-term public and private keys will not be compromised. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
+`espLifetime`<br/>*integer*  | The phase-2 lifetime of the security association in seconds
+`dpd`<br/>*boolean*    | A method to detect an unavailable Internet Key Exchange (IKE) peer.
```
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: Any options here?

File: source/includes/cloudstack/_s2s_vpns.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+`ikeDhGroup`<br/>*string*  | A public-key cryptography protocol which allows two parties to establish a shared secret over an insecure communications channel. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
+`ikeLifetime`<br/>*integer*  | The phase-1 lifetime of the security association in seconds.
+`espEncryptionAlgorithm`<br/>*string*  | Encapsulating Security Payload (ESP) algorithm within phase-2. The supported encryption algorithms are AES128, AES192, AES256, and 3DES.
+`espHashAlgorithm`<br/>*string*  | Encapsulating Security Payload (ESP) hash for phase-2. Supported hash algorithms are SHA1 and MD5.
+`espPerfectForwardSecrecy`<br/>*string*  |  Forward Secrecy (or PFS) is the property that ensures that a session key derived from a set of long-term public and private keys will not be compromised. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
+`espLifetime`<br/>*integer*  | The phase-2 lifetime of the security association in seconds
+`dpd`<br/>*boolean*    | A method to detect an unavailable Internet Key Exchange (IKE) peer. 
+`forceEncap`<br/>*boolean* | Force encapsulation for Nat Traversal
```
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: NAT

File: source/includes/cloudstack/_s2s_vpns.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+`name`<br/>*string* | The name of the site-to-site VPN
+`state`<br/>*string* | The state of the site-to-site VPN. Can be Connected, Pending or Disconnected. If disconnected, you can try to use the [reset](#reset-a-site-to-site-vpn) operation
+`vpcId`<br/>*UUID* | The VPC for which the site-to-site VPN was created.
+`gateway`<br/>*string*  | The gateway of the network you want to connect to. NOTE: you cannot use a gateway that has already been used by a site-to-site VPN in your environment
+`cidr`<br/>*string*  | Cidr of the network you want to connect to.
+`ipSecPsk`<br/>*string*  | IPSec pre-shared key. Mmust contain at least 10 alphanumeric characters.
+`ikeEncryptionAlgorithm`<br/>*string*  | The Internet Key Exchange (IKE) policy for phase-1. The supported encryption algorithms are AES128, AES192, AES256, and 3DES.
+`ikeHashAlgorithm`<br/>*string*  | The IKE hash for phase-1. The supported hash algorithms are SHA1 and MD5.
+`ikeDhGroup`<br/>*string*  | A public-key cryptography protocol which allows two parties to establish a shared secret over an insecure communications channel. The supported options are None, Group-5 (1536-bit) and Group-2 (1024-bit).
```
Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: currently, _None_ maps to null in the UI, so it's not a valid value. We should say that it's optional, with only 2 valid values when sent.

File: source/includes/cloudstack/_s2s_vpns.md (hunk 4 above)
Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: Same as dhGroup 👆

Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: it's the real name 😆

File: source/includes/cloudstack/_s2s_vpns.md (hunk 6 above)
Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: it's boolean

File: source/includes/cloudstack/_s2s_vpns.md (hunk 4 above)
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: Yes, but the property is esp**Perfect**ForwardSecrecy and *PFS*

File: source/includes/cloudstack/_s2s_vpns.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+`dpd`<br/>*boolean*    | A method to detect an unavailable Internet Key Exchange (IKE) peer. 
+`forceEncap`<br/>*boolean* | Force encapsulation for Nat Traversal
+
+Query Parameters | &nbsp;
+---------- | -----
+`vpc_id`<br/>*UUID* | Filter the list to only retrieve the site-to-site VPNs of a [VPC](#vpcs)
+
+<!-------------------- RETRIEVE A SNAPSHOT -------------------->
+
+#### Retrieve a site-to-site VPN
```
Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: Same comments about the attributes will apply here too

File: source/includes/cloudstack/_s2s_vpns.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+
+curl -X POST \
+   -H "Content-Type: application/json" \
+   -H "MC-Api-Key: your_api_key" \
+   "https://cloudmc_endpoint/v1/services/compute-on/test_area/sitetositevpns/ca86b14f-20db-463d-b58a-9d3fa5959af2?operation=reset"
+
+```
+ <code>POST /services/<a href="#service-connections">:service_code</a>/<a href="#environments">:environment_name</a>/sitetositevpns/:id?operation=reset</code>
+
+Reset a site-to-site VPN.
```
Comment (simongodard) [inline source/includes/cloudstack/_s2s_vpns.md]: Reset _the connection_ of a site-to-site VPN

File: source/includes/cloudstack/_s2s_vpns.md (hunk 6 above)
Comment (pdube) [inline source/includes/cloudstack/_s2s_vpns.md]: Ah missed that my bad

File: source/includes/cloudstack/_s2s_vpns.md (hunk 8 above)
Comment (franzpgarcia) [inline source/includes/cloudstack/_s2s_vpns.md]: Ah ok will do

File: source/includes/cloudstack/_s2s_vpns.md (hunk 9 above)
Comment (franzpgarcia) [inline source/includes/cloudstack/_s2s_vpns.md]: Done

File: source/includes/cloudstack/_s2s_vpns.md (hunk 10 above)
Comment (franzpgarcia) [inline source/includes/cloudstack/_s2s_vpns.md]: Done

File: source/includes/cloudstack/_s2s_vpns.md (hunk 5 above)
Comment (franzpgarcia) [inline source/includes/cloudstack/_s2s_vpns.md]: Done

File: source/includes/cloudstack/_s2s_vpns.md (hunk 1 above)
Comment (franzpgarcia) [inline source/includes/cloudstack/_s2s_vpns.md]: Done

**Pull request merged**